projects/packages/waf/src/class-waf-runner.php

```diff
@@ -151,10 +153,57 @@ public static function activate() {
 		if ( ! $version ) {
 			add_option( self::VERSION_OPTION_NAME, self::WAF_RULES_VERSION );
 		}
+		self::create_waf_directory();
+		self::create_blocklog_table();
 		self::generate_rules();
 	}
 
 	/**
+	 * Created the waf directory on activation.
+	 *
+	 * @return void
+	 * @throws \Exception In case there's a problem when creating the directory.
+	 */
+	public static function create_waf_directory() {
+		WP_Filesystem();
+		Waf_Constants::initialize_constants();
+
+		global $wp_filesystem;
+		if ( ! $wp_filesystem ) {
+			throw new \Exception( 'Can not work without the file system being initialized.' );
+		}
+
+		if ( ! $wp_filesystem->is_dir( JETPACK_WAF_DIR ) ) {
+			if ( ! $wp_filesystem->mkdir( JETPACK_WAF_DIR ) ) {
+				throw new \Exception( 'Failed creating WAF standalone bootstrap file directory: ' . JETPACK_WAF_DIR );
+			}
+		}
+	}
+
+	/**
+	 * Create the log table when plugin is activated.
+	 *
+	 * @return void
+	 */
+	public static function create_blocklog_table() {
+		global $wpdb;
+
+		require_once ABSPATH . 'wp-admin/includes/upgrade.php';
+
+		$sql = "
+		CREATE TABLE {$wpdb->prefix}jetpack_waf_blocklog (
+			log_id BIGINT UNSIGNED NOT NULL AUTO_INCREMENT,
+			timestamp datetime NOT NULL,
+			rule_id BIGINT NOT NULL,
+			reason longtext NOT NULL,
+			PRIMARY KEY (log_id),
+			KEY timestamp (timestamp)
+		)
+	";
+
+		dbDelta( $sql );
+  }
```

Comment:
Some merge issues here?

Reply:
Yep, tried to resolve them using the web interface and probably messed it up somehow. Fixed it.